Automattic/wp-calypso
# Add story for the features-grid on the /setup/newsletter user flow

Related to #88935

## Proposed Changes

With Storybook now in place for the `plans-grid-next` package, the plan is to add stories that correspond to each variation of the `FeaturesGrid` component ie. as it appears for each route / user flow. This PR focuses on adding a story for the `/setup/newsletter` user flow, with followup PR's addressing other `/setup/xx` flows.

Note that the features displayed for each plan are not accurate - this is fine for now, as it's currently just mock data. Once we have API request mocking in place (https://github.com/Automattic/wp-calypso/issues/89526) we'll have accurate data presented for each plan.

I've also updated/improved some of the TS types in our story definitions. For example each story is now typed based on the component: `type Story = StoryObj< typeof FeaturesGrid >`.

Here's the new `FeaturesGrid` story for the `/setup/newsletter` user flow: 

![CleanShot 2024-04-24 at 17 25 46](https://github.com/Automattic/wp-calypso/assets/69198925/a609ce04-025c-487a-9ded-996bad63bb83)

I've also updated the naming/hierarchy in the left-hand nav to be more intuitive:

| Before | After |
| -- | -- |
|![CleanShot 2024-04-24 at 17 27 33](https://github.com/Automattic/wp-calypso/assets/69198925/39f4199f-9a3c-411e-a5af-758991239801)|![CleanShot 2024-04-24 at 17 28 24](https://github.com/Automattic/wp-calypso/assets/69198925/ec561a83-bdc0-47a8-83e7-ff970029282c)




## Testing Instructions

1. Run `yarn workspace @automattic/plans-grid-next storybook`
2. Open the localhost URL for Storybook
3. Use the left-hand nav to view the new story: `FeaturesGrid > /setup/newsletter`
4. Check that the story renders

## Pre-merge Checklist
